### PR TITLE
Enhance PointerName rule analysis and documentation

### PR DESCRIPTION
**Pull request by denieu**
This PR improves the behavior of the PointerName rule for types that do not follow naming conventions. It also allows pointers to types with the prefix E and I, in addition to the already supported T.

Fixes https://github.com/integrated-application-development/sonar-delphi/issues/238

### CODE REVIEW

File: delphi-checks/src/main/java/au/com/integradev/delphi/checks/PointerNameCheck.java
Comment (fourls) [inline delphi-checks/src/main/java/au/com/integradev/delphi/checks/PointerNameCheck.java:45]: This can be done more concisely by constructing the list directly as the static final variable with `List.of`:

```suggestion
  private static final List<String> EXTENDED_TYPE_PREFIXES = List.of("T", "E", "I");
```

File: delphi-checks/src/main/java/au/com/integradev/delphi/checks/PointerNameCheck.java
Comment (fourls) [inline delphi-checks/src/main/java/au/com/integradev/delphi/checks/PointerNameCheck.java:57]: Now that there's more logic in this method, there's a lot of quite indented code - I wonder if it could be a good idea to change this `type.isPointer()` check to be an early return.

File: delphi-checks/src/main/java/au/com/integradev/delphi/checks/PointerNameCheck.java
Comment (fourls) [inline delphi-checks/src/main/java/au/com/integradev/delphi/checks/PointerNameCheck.java:75]: This currently raises an issue when custom prefixes are used, e.g.
```delphi
type PxyMyClass = ^TxyMyClass;
```

File: delphi-checks/src/test/java/au/com/integradev/delphi/checks/PointerNameCheckTest.java
Comment (fourls) [inline delphi-checks/src/test/java/au/com/integradev/delphi/checks/PointerNameCheckTest.java:38]: Little typo here.
```suggestion
                .appendDecl("  PCompliantPointer = ^noncompliant_type;")
```

File: delphi-checks/src/test/java/au/com/integradev/delphi/checks/PointerNameCheckTest.java
Comment (fourls) [inline delphi-checks/src/test/java/au/com/integradev/delphi/checks/PointerNameCheckTest.java:1]: Thanks for updating the check tests for this! It could also be good to have some new tests around custom prefixes, e.g.

Compliant names with custom prefixes:

```delphi
type
  PxyMyClass = ^TxyMyClass;
```

Prefix mismatches:

```delphi
type
  PMyClass = ^TxyMyClass; // Noncompliant
  PxyMyClass = ^TMyClass; // Noncompliant
```

File: delphi-checks/src/main/java/au/com/integradev/delphi/checks/PointerNameCheck.java
Comment (fourls) [inline delphi-checks/src/main/java/au/com/integradev/delphi/checks/PointerNameCheck.java:1]: Since the rule behaviour is changing, the PointerName rule description in [PointerName.html](https://github.com/integrated-application-development/sonar-delphi/blob/master/delphi-checks/src/main/resources/org/sonar/l10n/delphi/rules/community-delphi/PointerName.html) should also be updated.

**Review by fourls — CHANGES_REQUESTED** [review]
Hi @denieu, thanks for raising this PR!

Overall it looks great - tests pass, formatted, well set up, etc. I've added a couple of notes but they are mostly small things.

The main bug I've noticed is that custom prefixes aren't supported at all - this case we discussed gets erroneously raised as an issue:

```delphi
type PxyMyClass = ^TxyMyClass;
```

I still think that the logic we need can be summarised as:

A pointer type is the result of either

1. Removing the initial `T` / `E` / `I`
2. Adding a `P`

or

1. Adding a `P`

I think this catches all cases except `PxyMyClass = TMyClass`, but there's a fair argument to be made that that should raise an issue anyway, so I'm not particularly bothered about it.

Looking forward to seeing the next iteration 😄

**Review by fourls — CHANGES_REQUESTED** [review]
Thanks @denieu, the code change looks all good now.

Thanks for updating the rule description as well - it looks good overall, and I definitely agree that it should be detailed. That being said, it is probably a little bit long in its current state - I think that once it's trimmed a little it will be good to go.

I've left some notes about possible ways to make it more concise, but feel free to take different approaches if you have any other ideas!

**Comment by denieu** [comment]
Thanks for the review @fourls.

I made the suggested adjustments and also added a description of the changes made to CHANGELOG.md.

File: delphi-checks/src/main/resources/org/sonar/l10n/delphi/rules/community-delphi/PointerName.html
Comment (fourls) [inline delphi-checks/src/main/resources/org/sonar/l10n/delphi/rules/community-delphi/PointerName.html:5]: The end of this sentence doesn't quite make sense - it's probably okay to remove, since this information is described later on.

```suggestion
  <code>P</code>.
```
Rewording is also OK if you would prefer.

File: CHANGELOG.md
Comment (fourls) [inline CHANGELOG.md:26]: Thanks for updating the changelog! Usually a maintainer would do this before merging (to avoid merge conflicts).

I'd consider this an improvement rather than a fix, as it's making the rule behaviour meaningfully better rather than fixing a quirk. Because of this, I think it could be summed up in a single bullet point under the `Changed` heading instead of two separate points in different headings.

**Review by fourls — CHANGES_REQUESTED** [review]
Thanks @denieu, almost good to go now. Just a couple wording notes and that's it :smile:

**Comment by denieu** [comment]
@fourls I made the suggested adjustments and also changed the title of the commit and the PR to comply with the changes.

**Review by fourls — APPROVED** [review]
Looks good 🎉 Thanks for your effort in submitting this PR, it's a great quality of life improvement for PointerName.

The next official release will be in a week or two, so there's not long to wait until it's in everyone's hands!

**Pull request merged**
